lib/elixir/src/elixir_erl_pass.erl

```diff
@@ -12,8 +12,27 @@ translate({'=', Meta, [{'_', _, Atom}, Right]}, S) when is_atom(Atom) ->
 
 translate({'=', Meta, [Left, Right]}, S) ->
   {TRight, SR} = translate(Right, S),
-  {TLeft, SL} = elixir_erl_clauses:match(fun translate/2, Left, SR),
-  {{match, ?ann(Meta), TLeft, TRight}, SL};
+  case elixir_erl_clauses:match(fun translate/2, Left, SR) of
+    {TLeft, #elixir_erl{extra_guards=ExtraGuards, context=Context} = SL0}
+        when ExtraGuards =/= [], Context =/= match ->
+      SL1 = SL0#elixir_erl{extra_guards=[]},
+      Match = {match, ?ann(Meta), TLeft, TRight},
+      Generated = ?generated(Meta),
+      {ResultVar, SL2} = elixir_erl_clauses:match(fun translate/2, {result, Generated, ?var_context}, SL1),
```

Comment:
I have looked at it before, `elixir_erl_var:build/2` uses regular context, not `?var_context`. 🤔